helix-term/src/ui/popup.rs

```diff
@@ -124,16 +124,17 @@ impl<T: Component> Component for Popup<T> {
         // -- make sure frame doesn't stick out of bounds
         let mut rel_x = position.col as u16;
         let mut rel_y = position.row as u16;
-        if viewport.width <= rel_x + width {
-            rel_x -= ((rel_x + width) - viewport.width)
+        if viewport.width <= rel_x.saturating_add(width) {
```

Comment:
No need for the `saturating_add`s I think, just `sub`.

We should really audit all the `ui` components for these types of issues... Safest way seemed to be to use `viewport.intersection(area)` to clip into bounds

Reply:
> No need for the `saturating_add`s I think, just `sub`.
> 
> ... Safest way seemed to be to use `viewport.intersection(area)` to clip into bounds
This is already what happens few lines down:
```rust
        let area = Rect::new(rel_x, rel_y, width, height);
        // clip to viewport
        let area = viewport.intersection(Rect::new(rel_x, rel_y, width, height));
```

Reply:
Removed the saturating adds.